src/validators/StringValidator.ts

```diff
@@ -29,6 +40,27 @@ export class StringValidator<T extends string> extends BaseValidator<T> {
 		return this.addConstraint(stringLengthNe(length) as IConstraint<T>);
 	}
 
+	public email(): this {
+		return this.addConstraint(stringRegex(emailRegex, 'email') as IConstraint<T>);
+	}
+
+	public url(options?: UrlOptions): this {
+		return this.addConstraint(stringUrl(options) as IConstraint<T>);
+	}
+
+	public uuid(v: number | `${bigint}-${bigint}` | null = 4): this {
+		// from https://stackoverflow.com/a/38191078/16893382
+		const uuidRegex = new RegExp(
+			`^([0-9A-F]{8}-[0-9A-F]{4}-[${v ?? '1-5'}][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}|00000000-0000-0000-0000-000000000000)$`,
```

Comment:
Wait, why is there a null uuid?

Reply:
https://github.com/sapphiredev/shapeshift/pull/38#discussion_r813677328
ig I've misunderstood it.

Reply:
Yes, that comment was resolved correctly, `v` being `null` won't default to `4`, so it'll be defaulted to `1-5` in line 54.

However, a null UUID isn't... exactly what I meant, they're also not valid as well

Reply:
Oh, that `00` I've added the regex from the StackOverflow link but then as you'd asked that all 5 types of uuid have the same regex or not so I checked zod regex and found they have a null uuid. Idk why it's there and I though it's also a valid uuid so I added it. I'll remove it if u want.

Reply:
Now after searching a bit I found its a valid uuid.

Source https://en.m.wikipedia.org/wiki/Universally_unique_identifier
![Screenshot_20220224-194256_Chrome](https://user-images.githubusercontent.com/74945038/155540687-d7053213-deaa-445a-9c76-c9668203768c.jpg)

Reply:
Oh, interesting, I didn't know that, then it's fine.